grate/Migration/AnsiSqlDatabase.cs

```diff
@@ -288,9 +288,17 @@ protected virtual string ExistsSql(string tableSchema, string fullTableName)
 
     public async Task<string> GetCurrentVersion()
     {
-        var sql = CurrentVersionSql;
-        var res = await ExecuteScalarAsync<string>(Connection, sql);
-        return res ?? "0.0.0.0";
+        try
+        {
+            var sql = CurrentVersionSql;
+            var res = await ExecuteScalarAsync<string>(Connection, sql);
+            return res ?? "0.0.0.0";
+        }
+        catch (Exception ex)
```

Comment:
I know this is merged, so this is more of a general comment. I'm very sceptical to catching `Exception` in general, as this could be everything from `OutOfMemoryException` to `FileNotFoundException`, so I think we should narrow this catch a lot down, if possible. And, if it's always a generic `DbException` or sim, we could try to look a bit at the details to make sure it's actually. the exxpected error we are getting, so that we don't swallow any other exception by mistake :)

Reply:
Completely agree.  I can't even remember which exception type I was seeing when I wrote this!